### PR TITLE
Automatic port of Microsoft.Extensions.Dependency* docs

### PR DESCRIPTION
**Pull request by carlossanlop**
Area owners of Microsoft.Extensions.Dependency*: @maryamariyan @eerhardt

### CODE REVIEW

**Comment by eerhardt** [comment]
Tagging @dotnet/ncl as the owner of `HttpClientBuilderExtensions` and `HttpClientFactoryServiceCollectionExtensions`.

Tagging @ViktorHofer as the co-owner of `OptionsBuilderConfigurationExtensions`.

**Review by eerhardt — APPROVED** [review]
The `Microsoft.Extensions.DependencyModel` changes look good to me.

File: xml/Microsoft.Extensions.DependencyInjection/HttpClientFactoryServiceCollectionExtensions.xml [hunk 1]
```diff
@@ -724,13 +724,27 @@
...
+## Remarks
+
+ <xref:System.Net.Http.HttpClient> instances that apply the provided configuration can be retrieved using <xref:System.Net.Http.IHttpClientFactory.CreateClient(System.String)> and providing the matching name.
+            
+
+ <typeparamref name="TClient> instances constructed with the appropriate <xref:System.Net.Http.HttpClient> can be retrieved from <xref:System.IServiceProvider.GetService(System.Type)> (and related methods) by providing <typeparamref name="TClient> as the service type.
+            
+
+
+
```
Comment (carlossanlop) [inline xml/Microsoft.Extensions.DependencyInjection/HttpClientFactoryServiceCollectionExtensions.xml:745]:
```suggestion
```

File: xml/Microsoft.Extensions.DependencyInjection/HttpClientFactoryServiceCollectionExtensions.xml [hunk 2]
```diff
@@ -766,13 +780,27 @@
...
```
Comment (carlossanlop) [inline xml/Microsoft.Extensions.DependencyInjection/HttpClientFactoryServiceCollectionExtensions.xml:801]:
```suggestion
```

File: xml/Microsoft.Extensions.DependencyInjection/HttpClientFactoryServiceCollectionExtensions.xml [hunk 3]
```diff
@@ -766,13 +780,27 @@
...
+        <summary>Adds the <see cref="T:System.Net.Http.IHttpClientFactory" /> and related services to the <see cref="T:Microsoft.Extensions.DependencyInjection.IServiceCollection" /> and configures a binding between the <typeparamref name="TClient" /> type and a named <see cref="T:System.Net.Http.HttpClient" />.</summary>
+        <returns>An <see cref="T:Microsoft.Extensions.DependencyInjection.IHttpClientBuilder" /> that can be used to configure the client.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+ <xref:System.Net.Http.HttpClient> instances that apply the provided configuration can be retrieved using <xref:System.Net.Http.IHttpClientFactory.CreateClient(System.String)> and providing the matching name.
+            
+
```
Comment (carlossanlop) [inline xml/Microsoft.Extensions.DependencyInjection/HttpClientFactoryServiceCollectionExtensions.xml:796]:
```suggestion
```

File: xml/Microsoft.Extensions.DependencyInjection/HttpClientFactoryServiceCollectionExtensions.xml [hunk 4]
```diff
@@ -724,13 +724,27 @@
...
```
Comment (carlossanlop) [inline xml/Microsoft.Extensions.DependencyInjection/HttpClientFactoryServiceCollectionExtensions.xml:740]:
```suggestion
```

File: xml/Microsoft.Extensions.DependencyInjection/HttpClientFactoryServiceCollectionExtensions.xml [hunk 5]
```diff
@@ -1008,14 +1036,28 @@
...
```
Comment (carlossanlop) [inline xml/Microsoft.Extensions.DependencyInjection/HttpClientFactoryServiceCollectionExtensions.xml:1053]:
```suggestion
```

File: xml/Microsoft.Extensions.DependencyInjection/HttpClientFactoryServiceCollectionExtensions.xml [hunk 6]
```diff
@@ -1008,14 +1036,28 @@
...
+ <typeparamref name="TClient> instances constructed with the appropriate <xref:System.Net.Http.HttpClient> can be retrieved from <xref:System.IServiceProvider.GetService(System.Type)> (and related methods) by providing <typeparamref name="TClient> as the service type.
+            
+
+
```
Comment (carlossanlop) [inline xml/Microsoft.Extensions.DependencyInjection/HttpClientFactoryServiceCollectionExtensions.xml:1057]:
```suggestion
```

File: xml/Microsoft.Extensions.DependencyInjection/HttpClientFactoryServiceCollectionExtensions.xml [hunk 7]
```diff
@@ -1052,14 +1094,28 @@
...
+        <summary>Adds the <see cref="T:System.Net.Http.IHttpClientFactory" /> and related services to the <see cref="T:Microsoft.Extensions.DependencyInjection.IServiceCollection" /> and configures a binding between the <typeparamref name="TClient" /> type and a named <see cref="T:System.Net.Http.HttpClient" />.</summary>
+        <returns>An <see cref="T:Microsoft.Extensions.DependencyInjection.IHttpClientBuilder" /> that can be used to configure the client.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+ <xref:System.Net.Http.HttpClient> instances that apply the provided configuration can be retrieved using <xref:System.Net.Http.IHttpClientFactory.CreateClient(System.String)> and providing the matching name.
+            
+
```
Comment (carlossanlop) [inline xml/Microsoft.Extensions.DependencyInjection/HttpClientFactoryServiceCollectionExtensions.xml:1111]:
```suggestion
```

File: xml/Microsoft.Extensions.DependencyInjection/HttpClientFactoryServiceCollectionExtensions.xml [hunk 8]
```diff
@@ -1052,14 +1094,28 @@
...
+ <typeparamref name="TClient> instances constructed with the appropriate <xref:System.Net.Http.HttpClient> can be retrieved from <xref:System.IServiceProvider.GetService(System.Type)> (and related methods) by providing <typeparamref name="TClient> as the service type.
+            
+
```
Comment (carlossanlop) [inline xml/Microsoft.Extensions.DependencyInjection/HttpClientFactoryServiceCollectionExtensions.xml:1114]:
```suggestion
```

File: xml/Microsoft.Extensions.DependencyModel/RuntimeAssetGroup.xml [hunk 9]
```diff
@@ -110,7 +110,7 @@
         <ReturnType>System.Collections.Generic.IReadOnlyList&lt;System.String&gt;</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Gets a list of asset paths provided in this runtime group</summary>
         <value>To be added.</value>
```
Comment (carlossanlop) [inline xml/Microsoft.Extensions.DependencyModel/RuntimeAssetGroup.xml:114]:
```suggestion
        <summary>Gets a list of asset paths provided in this runtime group.</summary>
        <value>A read-only list of strings that represent the asset paths.</value>
```

File: xml/Microsoft.Extensions.DependencyModel/RuntimeAssetGroup.xml [hunk 10]
```diff
@@ -131,7 +131,7 @@
         <ReturnType>System.String</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>The runtime ID associated with this group (may be empty if the group is runtime-agnostic)</summary>
         <value>To be added.</value>
```
Comment (carlossanlop) [inline xml/Microsoft.Extensions.DependencyModel/RuntimeAssetGroup.xml:135]:
```suggestion
        <summary>The runtime ID associated with this group (may be empty if the group is runtime-agnostic).</summary>
        <value>A string that represents the runtime ID.</value>
```

File: xml/Microsoft.Extensions.DependencyModel/RuntimeAssetGroup.xml [hunk 11]
```diff
@@ -152,7 +152,7 @@
         <ReturnType>System.Collections.Generic.IReadOnlyList&lt;Microsoft.Extensions.DependencyModel.RuntimeFile&gt;</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Gets a list of RuntimeFiles provided in this runtime group</summary>
         <value>To be added.</value>
```
Comment (carlossanlop) [inline xml/Microsoft.Extensions.DependencyModel/RuntimeAssetGroup.xml:156]:
```suggestion
        <summary>Gets a list of runtime files provided in this runtime group.</summary>
        <value>A read-only list of runtime file instances.</value>
```

File: xml/Microsoft.Extensions.DependencyModel/RuntimeLibrary.xml [hunk 12]
```diff
@@ -123,20 +123,27 @@
...
+        <param name="runtimeStoreManifestName">The library's runtime store manifest name.</param>
+        <summary>Initializes a new <see cref="T:Microsoft.Extensions.DependencyModel.RuntimeLibrary" />.</summary>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentNullException">The <paramref name="type" /> argument is <see langword="null" />.
+            The <paramref name="name" /> argument is <see langword="null" />.
+            The <paramref name="version" /> argument is <see langword="null" />.
+            The <paramref name="runtimeAssemblyGroups" /> argument is <see langword="null" />.
+            The <paramref name="nativeLibraryGroups" /> argument is <see langword="null" />.
+            The <paramref name="resourceAssemblies" /> argument is <see langword="null" />.
+            The <paramref name="dependencies" /> argument is <see langword="null" />.</exception>
```
Comment (carlossanlop) [inline xml/Microsoft.Extensions.DependencyModel/RuntimeLibrary.xml:146]:
```suggestion
        <exception cref="T:System.ArgumentNullException"><paramref name="type" />, or <paramref name="name" />, or <paramref name="version" />, or <paramref name="runtimeAssemblyGroups" />, or <paramref name="nativeLibraryGroups" />, or <paramref name="resourceAssemblies" /> or <paramref name="dependencies" /> is <see langword="null" />.</exception>
```

**Review by maryamariyan — APPROVED** [review]

**Pull request merged**